### PR TITLE
Use reworked helmfile setup

### LINKED ISSUE #1919: Integrate new helmfile setup as default
**Is your feature request related to a problem? Please describe.**

With some testing done we are now ready to fully integrate the new setup into apps.

**Describe the solution you'd like**

- Remove old helmfile state files, move the root `helmfile.yaml` into `helmfile/state.yaml`.

- Rework paths to resole from within the `helmfile/` directory.

- Clean up old references

- Update the documentation to reflect the changes.

**Definition of done:**

All above steps have been completed.

### PR DESCRIPTION
**Pull request by aarnq**
> [!warning]
> **This is public repository, ensure not to disclose:**
> - [x] personal data beyond what is necessary for interacting with this pull request
> - [x] business confidential information, such as customer names

### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [ ] kind/feature       
- [x] kind/improvement   
- [ ] kind/deprecation   
- [ ] kind/documentation 
- [ ] kind/clean-up      
- [ ] kind/bug           
- [ ] kind/other         

*Optional*: Mark one or more of the following that are applicable:

> [!important]
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type
> Critical security fixes should be marked with `kind/security`

- [x] kind/admin-change  
- [ ] kind/dev-change    
- [ ] kind/security      
- [ ] [kind/adr]()       

### Platform Administrator notice

- The new helmfile setup is now standard and used by the bin scripts, use `./bin/ck8s ops helmfile sc|wc list` to view changed labels, it will show all releases in the list however only enabled ones will be managed for that cluster.

### What does this PR do / why do we need this PR?

Use the new helmfile setup as default.

Also renamed the dir `helmfile` -> `helmfile.d` which allows `helmfile` to be invoked from the root of the repo without targeting a particular file.

- Fixes #1919

#### Checklist

- [x] Proper commit message prefix on all commits
  
- Change checks:
  - [x] The change is transparent
  - [ ] The change is disruptive
  - [x] The change requires no migration steps
  - [ ] The change requires migration steps
- Metrics checks:
  - [ ] The metrics are still exposed and present in Grafana after the change
  - [ ] The metrics names didn't change (Grafana dashboards and Prometheus alerts are not affected)
  - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts were fixed)
- Logs checks:
  - [ ] The logs do not show any errors after the change
- Network Policy checks:
  - [ ] Any changed pod is covered by Network Policies
  - [ ] The change does not cause any dropped packages in the `NetworkPolicy Dashboard`
- Pod Security Policy checks:
  - [ ] Any changed pod is covered by Pod Security Admission
  - [ ] Any changed pod is covered by Gatekeeper Pod Security Policies
  - [ ] The change does not cause any pods to be blocked by Pod Security Admission or Policies
- Falco checks:
  - [ ] The change does not cause any alerts to be generated by Falco
- Audit checks:
  - [ ] The change does not cause any unnecessary Kubernetes audit events
  - [ ] The change requires changes to Kubernetes audit policy
- Bug checks:
  - [ ] The bug fix is covered by regression tests

### CODE REVIEW

File: helmfile.d/README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+Compliant Kubernetes Apps is driven by [`helmfile`](https://github.com/helmfile/helmfile).
+
+The structure is as follows:
+
+- `bases/` - contains the environment definition and some common values
+- `charts/` - contains internally developed charts grouped per stack
+- `upstream/` - contains externally developed charts grouped per source
+- `stacks/` - contains release templates for all applications grouped per stack
+- `values/` - contains values templates for all applications grouped per stack
```
Comment (OlleLarsson) [inline helmfile.d/README.md:11]: I suppose this is more correct
```suggestion
- `values/` - contains values templates for most applications
```

File: helmfile.d/README.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+- `state.yaml` - the state file that pulls in everything and defines releases
+
+## Usage
+
+While `helmfile` can be invoked via `./bin/ck8s ops helmfile <sc|wc>` it can also be invoked directly from the root of the repository.
+Note that you must then point `KUBECONFIG` to the correct cluster and invoke it as `helmfile -e service_cluster|workload_cluster`.
+
+Selective operations can be done using `key=value` labels, use the `helmfile` operation `list` to view them.
```
Comment (OlleLarsson) [inline helmfile.d/README.md:19]: Maybe include the flag used to select based on key/values
```suggestion
Selective operations can be done using `-l key=value` labels, use the `helmfile` operation `list` to view them.
```

File: helmfile.d/README.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+- `charts/` - contains internally developed charts grouped per stack
+- `upstream/` - contains externally developed charts grouped per source
+- `stacks/` - contains release templates for all applications grouped per stack
+- `values/` - contains values templates for all applications grouped per stack
+- `state.yaml` - the state file that pulls in everything and defines releases
+
+## Usage
+
+While `helmfile` can be invoked via `./bin/ck8s ops helmfile <sc|wc>` it can also be invoked directly from the root of the repository.
+Note that you must then point `KUBECONFIG` to the correct cluster and invoke it as `helmfile -e service_cluster|workload_cluster`.
```
Comment (OlleLarsson) [inline helmfile.d/README.md:17]: I would add an important note here saying that there is no config version validation when using plain helmfile so one needs to be careful!

File: helmfile.d/README.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+> [!warning]
+> When using `helmfile` directly there is no validation done so be careful!
+
+Selective operations can be done using the `---selector|-l` argument followed by `key=value` labels, additional `--selector|-l` performs or, additional labels delimited by comma performs and.
```
Comment (OlleLarsson) [inline helmfile.d/README.md:22]:
```suggestion
Selective operations can be done using the `---selector|-l` argument followed by `key=value` labels, additional `--selector|-l` performs _or_, additional labels delimited by comma performs _and_.
```

**Review by OlleLarsson — APPROVED** [review]

File: release/README.md [hunk 5]
```diff
@@ -98,8 +98,8 @@ Write about some new feature or change that is relevant for the user, e.g. for `
 
 Remove the items in this section from two+ older minor versions, meaning if you release apps `v0.28` you can keep previous items that were added to the list in `v0.27` but remove the stuff that are from `v0.26`.
 
-- Edit the [Grafana dashboard](../helmfile/charts/grafana-ops/files/welcome.md)
-- Edit the [Opensearch dashboard](../helmfile/charts/opensearch/configurer/files/dashboards-resources/welcome.md)
+- Edit the [Grafana dashboard](../helmfile.d/charts/grafana-ops/files/welcome.md)
```
Comment (davidumea) [inline release/README.md:101]:
```suggestion
- Edit the [Grafana dashboard](../helmfile.d/charts/grafana-dashboards/files/welcome.md)
```

File: helmfile.d/stacks/ingress-nginx.yaml [hunk 6]
```diff
@@ -5,14 +5,24 @@ templates:
...
+  file-copier:
+    inherit: [ template: ingress-nginx ]
+    name: file-copier
+    namespace: kube-system
+    chart: charts/file-copier
+    version: 0.1.0
+    installed: {{ .Values | get "ingressNginx.controller.chroot" false }}
+    values:
+      - values/file-copier.yaml.gotmpl
+
```
Comment (davidumea) [inline helmfile.d/stacks/ingress-nginx.yaml:17]: I was thinking that "file-copier" as a chart could be separated from ingress-nginx, if we want to copy more files onto hosts (e.g. public ssh keys as you mentioned) we could just put more daemonsets/configmaps in it.

File: DEVELOPMENT.md
Comment (viktor-f) [inline DEVELOPMENT.md:1]: Are there any specific version requirements on kind and docker/podman?
If there is then mention that in this doc. If there is not, then consider if there should be a note on which versions we have verified that it works for. I don't really know if we would expect any changes to impact us.

File: DEVELOPMENT.md [hunk 7]
```diff
@@ -86,6 +86,28 @@ helmfile -e <service|workload>_cluster -lapp=ingress-nginx -lapp=node-local-dns
...
+harbor         | 🟩️ | ⬜️ | Does not pull in dex or ingress-nginx.
+hnc            | ⬜️ | 🟩️ |
+ingress-nginx  | 🟩️ | 🟩️ |
+kured          | 🟩️ | 🟩️ |
+node-local-dns | 🟩️ | 🟩️ |
+opensearch     | 🟩️ | ⬜️ | Does not pull ingress-nginx. Prod flavour is heavy on resources.
+prometheus     | 🟩️ | 🟩️ |
+thanos         | 🟩️ | ⬜️ | Does not pull cert-manager, ingress-nginx.
+trivy-operator | 🟩️ | 🟩️ |
+velero         | 🟩️ | 🟩️ |
```
Comment (viktor-f) [inline DEVELOPMENT.md:109]: Very nice matrix :slightly_smiling_face: 
Nit: you could add definitions on what the colors represent. For me they are self-explanatory, but I could see someone being confused about white in particular. Add that if you want.

Comment (viktor-f) [reply DEVELOPMENT.md:109]: Questions on support for some minor things in apps:
Does it support alertmanager and calicoaccountant?

Comment (viktor-f) [reply DEVELOPMENT.md:109]: In the demo you showed me we talked about prometheus rules and that when you just install grafana you don't get all rules. You could add something in the grafana info about which releases to add in order to get all metrics.

**Review by viktor-f — COMMENTED** [review]
Looks nice, will try to do a quick test during the day as well.

File: DEVELOPMENT.md (hunk 7 above)
Comment (aarnq) [reply DEVELOPMENT.md:109]: Calico includes calico-accountant, Prometheus includes alertmanager. so they work. :) Will add notes about dependencies not pulling in resources :+1:

File: DEVELOPMENT.md
Comment (aarnq) [reply DEVELOPMENT.md:1]: Not that I'm aware of. As long as kind works as it should then docker and podman version doesn't matter. And for kind you want a recent enough version that the Kubernetes version still is relevant.

File: helmfile.d/stacks/ingress-nginx.yaml (hunk 6 above)
Comment (aarnq) [reply helmfile.d/stacks/ingress-nginx.yaml:17]: Yes, I'm thinking we'll move it to system once it is more generic.

File: DEVELOPMENT.md (hunk 7 above)
Comment (aarnq) [reply DEVELOPMENT.md]: Added predefined selectors as a guide.

**Review by viktor-f — APPROVED** [review]
I did a quick test and it worked well. Very well done with all the work to be able to run this on kind!

As a sidenote I found a small bug where I could not log in to the harbor I deployed (tried the admin used but it just told me I had the wrong username/password). Something to look at later. I just port-forwarded to the portal pod, so I did not have nginx.

**Pull request merged**